### PR TITLE
Allow products to be created independent of cases

### PR DESCRIPTION
**Pull request by stdstaff**
https://trello.com/c/EP4zMrDf/1572-create-a-product-record-not-linked-to-a-case-cap102-cap103-rules

- New routes and controller actions for creating a product outside of a case (`/products/new`)
- New CreateProduct service
- New confirmation page when a product is created
- Added a TODO at the old route (`/cases/xxxx-xxxx/products/new`) as a number of places link there, and a new user journey is being developed as part of the `new-case-creation-flow-phase-2` feature branch
- Added links to create a product to the product search pages

### CODE REVIEW

File: app/views/products/confirmation.html.erb [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+      </span>
+      <span class="govuk-!-display-inline-block govuk-!-margin-bottom-4">
+        You create a case for a product directly from its product record page.
+      </span>
+      <span class="govuk-!-display-inline-block">
+        On the <%= link_to "product record page", @product, class: "govuk-link govuk-link--no-visited-state" %> select 'Create a new case for this product'.
+      </span>
+    </div>
+
+    <p class="govuk-body-s">By using the <abbr>PSD</abbr> reference number you can also add this product record to pre-existing cases. You add a product record to a case directly from the case page.</p>
```
Comment (slorek) [inline app/views/products/confirmation.html.erb:24]:
```suggestion
    <p class="govuk-body-s">By using the <abbr title="Product Safety Database">PSD</abbr> reference number you can also add this product record to pre-existing cases. You add a product record to a case directly from the case page.</p>
```

File: app/views/products/heading/_all_products.html.erb [hunk 2]
```diff
@@ -1,8 +1,6 @@
 <div class="govuk-grid-row">
   <div class="govuk-grid-column-two-thirds opss-desktop-min-height--s">
-    <h1 class="govuk-heading-l govuk-!-margin-bottom-2">
-        All products – Search
-    </h1>
+    <h1 class="govuk-heading-l govuk-!-margin-bottom-2">All products – Search</h1>
     <p class="govuk-body opss-secondary-text">
       Search all product records in the Product Safety Database (<abbr>PSD</abbr>).
```
Comment (slorek) [inline app/views/products/heading/_all_products.html.erb:5]:
```suggestion
      Search all product records in the Product Safety Database (<abbr title="Product Safety Database">PSD</abbr>).
```

File: db/seeds.rb [hunk 3]
```diff
@@ -68,7 +68,8 @@ def create_blob(filename, title: nil, description: nil)
     has_markings: "markings_unknown"
   }
 
-  AddProductToCase.call!(product_params.merge({ investigation:, user: }))
+  product = CreateProduct.call!(product_params.merge({ user: })).product
+  investigation.products << product
```
Comment (slorek) [inline db/seeds.rb:72]: Suggest adding a `TODO` comment here as we'll want to use the new `AddProductToCase` service when that's build.

**Review by slorek — APPROVED** [review]

File: db/seeds.rb (hunk 3 above)
Comment (stdstaff) [reply db/seeds.rb:73]: https://github.com/OfficeForProductSafetyAndStandards/product-safety-database/pull/2062/commits/bae080cf3fda7c5e78e9f34e58daf02d0382f06a

**Pull request merged**